spark-connect/client/src/test/scala-spark-master/io/delta/connect/tables/RemoteSparkSession.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This file contains code from the Apache Spark project (original license above).
+ * It contains modifications, which are licensed as follows:
+ */
+
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.tables
+
+import org.scalatest.{BeforeAndAfterAll, Suite}
+
+import org.apache.spark.sql.SparkSession
+
+/**
+ * An util class to start a local Delta Connect server in a different process for local E2E tests.
+ * Pre-running the tests, the Delta Connect artifact needs to be built using e.g. `build/sbt
+ * assembly`. It is designed to start the server once but shared by all tests. It is equivalent to
+ * use the following command to start the connect server via command line:
+ *
+ * {{{
+ * bin/spark-shell \
+ * --jars `ls spark-connect/server/target/**/delta-connect-server-assembly*SNAPSHOT.jar | paste -sd ',' -` \
+ * --conf spark.plugins=org.apache.spark.sql.connect.SparkConnectPlugin
+ * --conf spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension
+ * --conf spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog
+ * }}}
+ *
+ * Set system property `delta.test.home` or env variable `DELTA_HOME` if the test is not executed
+ * from the Delta project top folder.
+ */
+trait RemoteSparkSession extends BeforeAndAfterAll { self: Suite =>
+
+  private val serverPort = 15003
+  var spark: SparkSession = _
+
+  private val buildLocation = System.getProperty("delta.test.home")
+  private val deltaConnectJar = s"$buildLocation/" +
+    "spark-connect/server/target/scala-2.13/delta-connect-server-assembly-3.3.0-SNAPSHOT.jar"
```

Comment:
1. Because of https://github.com/delta-io/delta/blob/master/version.sbt 

"The version.sbt file is used in Scala projects to specify the version of the project. The version number is used when publishing the project's artifacts to a repository. It can also be used by other projects that depend on this project to specify which version they require."

2. We would have to use this [findJar](https://github.com/apache/spark/blob/master/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/RemoteSparkSession.scala#L66) function, which we can't call to https://github.com/apache/spark/blob/master/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/IntegrationTestUtils.scala, and copying the whole thing would create too much duplication I think.

Maybe we can remove the `private` in a bunch of these files https://github.com/delta-io/delta/tree/master/spark/src/main/scala/io/delta/tables, not sure if we have the time right now.

Reply:
> 1. Because of https://github.com/delta-io/delta/blob/master/version.sbt
> 
> "The version.sbt file is used in Scala projects to specify the version of the project. The version number is used when publishing the project's artifacts to a repository. It can also be used by other projects that depend on this project to specify which version they require."
> 

Ah right. I incorrectly though the next version is 3.4 not 3.3.

> 2. We would have to use this [findJar](https://github.com/apache/spark/blob/master/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/RemoteSparkSession.scala#L66) function, which we can't call to https://github.com/apache/spark/blob/master/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/IntegrationTestUtils.scala, and copying the whole thing would create too much duplication I think.
> 
> Maybe we can remove the `private` in a bunch of these files https://github.com/delta-io/delta/tree/master/spark/src/main/scala/io/delta/tables, not sure if we have the time right now.

For now you could add a TODO in the code to address later.